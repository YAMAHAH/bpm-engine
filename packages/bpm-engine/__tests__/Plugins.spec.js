import BPMEngine from 'bpm-engine';

import TaskManager from './Plugins/TaskManager';
import ExternalServices from './Plugins/ExternalServices';

describe('Plugins', () => {
  it('Can setup a UserTask Plugin', async () => {
    const taskManager = new TaskManager();
    const bpm = new BPMEngine({
      plugins: [taskManager],
    });
    expect(bpm.plugins.userTask).toMatchSnapshot();
  });

  it('Can setup a ServiceTask Plugin', async () => {
    const externalServices = new ExternalServices();
    const bpm = new BPMEngine({
      plugins: [externalServices],
    });
    expect(bpm.plugins.serviceTask).toMatchSnapshot();
  });
});
