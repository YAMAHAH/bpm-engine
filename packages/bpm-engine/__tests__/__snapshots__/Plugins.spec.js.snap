// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugins Can setup a ServiceTask Plugin 1`] = `
Object {
  "0": ExternalServices {
    "onActive": [Function],
  },
}
`;

exports[`Plugins Can setup a UserTask Plugin 1`] = `
Object {
  "0": TaskManager {
    "onActive": [Function],
  },
}
`;
